src/Data/Nat/Show.agda

```diff
@@ -8,29 +8,104 @@
 
 module Data.Nat.Show where
 
-open import Data.Nat
+open import Data.Char as Char using (Char)
+open import Data.Digit using (showDigit; toDigits)
+open import Function   using (_∘_; _$_)
+open import Data.List  using (List; []; _∷_; map; reverse)
+open import Data.Nat   using (ℕ; _≟_; suc; pred; _+_; _*_; _<_; _>_; z≤n; s≤s;
+                                                                        _≤?_)
+open import Data.Nat.DivMod     using (_div_; _%_; a≡a%n+[a/n]*n)
+open import Data.Product        using (proj₁)
+open import Data.Nat.Properties using
+                     (<-irrefl; m≢0⇒suc[pred[m]]≡m; +-comm; *-identityʳ;
+                                         m≤m+n; *-monoʳ-<; module ≤-Reasoning)
+open import Data.String as Str using (String)
+import      Induction.Nat
+open import Induction.WellFounded using (Acc)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl; sym;
+                                                         cong)
+open import Relation.Nullary           using (yes; no)
 open import Relation.Nullary.Decidable using (True)
-open import Data.String.Base as String using (String)
-open import Data.Digit
-open import Data.Product using (proj₁)
-open import Function
-open import Data.List.Base
+open import Relation.Nullary.Negation  using (contradiction)
 
--- showInBase b n is a string containing the representation of n in
--- base b.
 
+------------------------------------------------------------------------------
+open ≤-Reasoning
+
+open Acc using (acc)
+
+<-acc :  {n : ℕ} → Acc _<_ n
+                          -- This means that _<_ is well-founded on ℕ
+                          -- ("each number in ℕ is accessible from 0 by _<_").
+<-acc {n} = Induction.Nat.<-wellFounded n
+
+
+-- Conversion from unary representation to the representation by the given
+-- base. 
+toDigitNats : (base : ℕ) → base > 1 → ℕ → List ℕ
+toDigitNats 0             ()
+toDigitNats 1             1>1    _ =  contradiction 1>1 (<-irrefl refl)
+toDigitNats (suc (suc b)) base>1 x =  aux x <-acc [] 
+  where
+  base = suc (suc b);  pbase = pred base
+
+  aux : (n : ℕ) → Acc _<_ n → List ℕ → List ℕ 
+  aux 0       _        =  (0 ∷_)
+  aux (suc n) (acc wf) =  aux'
+    where
+    n' = suc n;   q = n' div base;   r = n' % base;   pq = pred q
+
+    n'≡r+q*base :  n' ≡ r + q * base
+    n'≡r+q*base =  a≡a%n+[a/n]*n n' pbase
+
+    aux' : List ℕ → List ℕ
+    aux' with q ≟ 0
+    ... | yes _   =  (r ∷_)
+    ... | no q≢0 =  aux q (wf _ q<n') ∘ (r ∷_)   -- use  n' ≡ r + q*base 
+      where  
+      suc-pq≡q = m≢0⇒suc[pred[m]]≡m q≢0
+      q<q*base  =
```

Comment:
This is a very general result and therefore should be added as a new proof in `Data.Nat.Properties` under the name:
`m<m*n : ∀ {m n} → 0 < m → 1 < n → m < m * n`. This will make the proof much shorter.

Reply:
1) Need it to be a separate PR ?

2) The name  ``m<m*n`` shows that you tend to agree with my style for naming (with skipping the name parts describing evident conditions).
I like better my style.
But a) I do not decide here,  b) Probably standard library needs to follow one line.
What is this line?